src/klein/_app.py

```diff
@@ -146,15 +145,29 @@ def __get__(self, instance, owner):
         if instance is None:
             return self
 
-        k = self._bound_klein_instances.get(instance)
+        if self._boundAs is None:
+            for name in dir(owner):
+                obj = getattr(owner, name)
+                if obj is self:
+                    self._boundAs = name
+                    break
+            else:
+                self._boundAs = 'unknown_' + str(id(self))
```

Comment:
This isn't covered but it should be.  It's probably easiest to hit this branch by wrapping a `Klein` instance in a proxy that causes `obj is self` above to fail.